src/operators/delay.ts

```diff
@@ -96,7 +96,8 @@ class DelaySubscriber<T> extends Subscriber<T> {
       const delay = Math.max(0, queue[0].time - scheduler.now());
       this.schedule(state, delay);
     } else {
-      source.active = false;
```

Comment:
I think the fix is a one-liner and `source.remove(this);` should be added here.

Reply:
Good point, I've totally missed that. Will fix the PR. Thanks.